src/components/dropdown/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { useBackdropClick } from "@/hooks/use-backdrop-click";
+import clsx from "clsx";
+import { ReactNode, useState } from "react";
+
+type Alignment = "left" | "right" | "fill";
+
+interface DropdownOption {
+  label: string;
+  value: string;
+}
+
+interface Props {
+  anchor: ReactNode;
+  options: DropdownOption[] | string[];
+  gap?: number;
+  alignment?: Alignment;
+  alignmentOffset?: number;
+  onSelect: (option: DropdownOption | string) => void;
+}
+
+export default function Dropdown({
+  anchor,
+  options,
+  gap = 8,
+  alignment = "left",
+  alignmentOffset = 0,
+  onSelect,
+}: Props) {
+  const [isOpen, setIsOpen] = useState(false);
+  const targetRef = useBackdropClick<HTMLDivElement>({
+    callback: () => setIsOpen(false),
+  });
+
+  const alignmentStyles = { top: `calc(100% + ${gap}px)` };
+  switch (alignment) {
+    case "left":
+      Object.assign(alignmentStyles, { left: alignmentOffset });
+      break;
+    case "right":
+      Object.assign(alignmentStyles, { right: alignmentOffset });
+      break;
+    case "fill":
+      Object.assign(alignmentStyles, { left: 0, right: 0 });
+      break;
+  }
+
+  const handleAnchorClick = () => {
+    setIsOpen(!isOpen);
```

Comment:
문제는 없겠지만 이렇게 하면 조금 더 안전하지 않을까 생각합니다
```suggestion
    setIsOpen((prev) => !prev);
```

Reply:
- `isOpen` 값은 비동기적으로 사용되지 않아서 직접 state값을 설정해도 안전합니다.
- 이런 상황에서 `(prev) => !prev` 함수를 사용하면 불필요하게 장황해진다고 느껴지는데요. 혹시 상태를 변경할 때 항상 함수를 통해 이전 값을 바탕으로 변경한다는 컨벤션을 추가하고 싶다면 거기에 맞춰도 좋을 것 같습니다.

Reply:
> isOpen 값은 비동기적으로 사용되지 않아서 직접 state값을 설정해도 안전합니다.

아하 그렇네요. 비동기적으로 사용될 일이 없으면 기존에 작업하신 게 더 깔끔할 것 같습니다!